scripts/tests/yamltests/parser.py

```diff
@@ -0,0 +1,716 @@
+            for key in value:
+                mapped_value[key] = self._config_variable_substitution(value[key])
+            return mapped_value
+        elif type(value) is str:
+            # For most tests, a single config variable is used and it can be replaced as in.
+            # But some other tests were relying on the fact that the expression was put 'as if' in
+            # the generated code and was resolved before being sent over the wire. For such
+            # expressions (e.g 'myVar + 1') we need to compute it before sending it over the wire.
+            tokens = value.split()
+            if len(tokens) == 0:
+                return value
+
+            substitution_occured = False
+            for idx, token in enumerate(tokens):
+                if token in self._runtime_config_variable_storage:
+                    variable_info = self._runtime_config_variable_storage[token]
+                    if type(variable_info) is dict and 'defaultValue' in variable_info:
+                        variable_info = variable_info['defaultValue']
+                    tokens[idx] = variable_info
+                    substitution_occured = True
+
+            if len(tokens) == 1:
+                return tokens[0]
+
+            tokens = [str(token) for token in tokens]
+            value = ' '.join(tokens)
+            # TODO we should move away from eval. That will mean that we will need to do extra
+            # parsing, but it would be safer then just blindly running eval.
+            return value if not substitution_occured else eval(value)
+        else:
+            return value
+
+
+class YamlTests:
+    _tests: list[_TestStepWithPlaceholders]
+    _index: 0
+
+    count: 0
+
+    def __init__(self, parsing_config_variable_storage: dict, definitions, tests: dict):
+        self._parsing_config_variable_storage = parsing_config_variable_storage
+        enabled_tests = []
+        for test in tests:
+            test_with_placeholders = _TestStepWithPlaceholders(
+                test, self._parsing_config_variable_storage, definitions)
+            if test_with_placeholders.is_enabled:
+                enabled_tests.append(test_with_placeholders)
+        fixes.try_update_yaml_node_id_test_runner_state(
+            enabled_tests, self._parsing_config_variable_storage)
+
+        self._runtime_config_variable_storage = copy.deepcopy(parsing_config_variable_storage)
+        self._tests = enabled_tests
+        self._index = 0
+        self.count = len(self._tests)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self) -> TestStep:
+        if self._index < self.count:
+            test = self._tests[self._index]
+            test_step = TestStep(test, self._runtime_config_variable_storage)
+            self._index += 1
+            return test_step
+
+        raise StopIteration
+
+
+class TestParser:
+    name: None
+    PICS: None
+    tests: None
+    _parsing_config_variable_storage: dict = {}
```

Comment:
This is not a standard way of expressing fields. Please have `__init__`  method declare the instance variable. These here are class variables and thus are global across instances of that class. Putting them at the top, and then overriding them in `__init__` is not needed and confusing. Please do this for all classes